### PR TITLE
Use GitHub actions

### PR DESCRIPTION
**Pull request by kiskoza**
I was doing some maintanence tasks for an other gem and when I added the new Ruby 3.3 to the CI pipeline, I saw a few deprecation warnings coming from this gem (as one of our dependencies). I saw that there's a PR already to get rid of the warning (#75), but the project lacks of a working pipeline to verify that change (travis seems to be unavailable). Looking at the open pull requests, there were others trying to make sure that the gem works with recent Ruby versions (#74, #71).

Fast-forward to my PR: I'm trying to add Github Actions as it seems to be a reliable pipeline for open source repositories hosted on Github.

The changes I made:
* replaced `.travis.yml` with `.github/workflows/test.yml`
* dropped Ruby 1.8 from the CI: the ubuntu-20.04 image does not support installing it - I can look into if we can get it work on other base OS, but it's EOL for ages
* dropped Ruby 1.9 from the CI: I wasn't able to setup the required gems, there's no compatible rexml version on rubygems: https://github.com/ruby/rexml/issues/38
* included all 2.x and 3.x Ruby versions and ruby-head as allowed failure
* included JRuby 9.3, 9.4 and jruby-head as allowed failure
* added some Gemfile constrains for Ruby < 2.1 to make sure it passes the CI
* added a Rakefile to make it easier to run the tests. we should convert the scripts folder to be rake tasks as well later
* cherry-picked a commit from #74 to make the CI pass for recent Ruby versions

I hope you'll find this PR helpful. Let me know if you need anything else before you can merge it in.

(edit) And example of a green pipeline could be found here https://github.com/kiskoza/crack/actions/runs/7384963412

### CODE REVIEW

File: lib/crack/json.rb [hunk 1]
```diff
@@ -4,20 +4,32 @@
 # THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 require 'strscan'
+require 'psych'
```
Comment (bf4) [inline lib/crack/json.rb:7]: I don't believe this change is in scope with ths pr

**Review by bf4 — CHANGES_REQUESTED** [review]
Please remove out of scope changes

File: lib/crack/json.rb (hunk 1 above)
Comment (kiskoza) [reply lib/crack/json.rb:7]: if I remember correctly, I had to add this to fix a failing test with one of the Ruby versions. I'm going to check it again and come back with the results

File: .travis.yml [hunk 2]
```diff
@@ -1,14 +0,0 @@
...
-  - 1.9
-  - 2.0
-  - 2.1
-  - 2.2
-  - jruby
-  - rbx
-matrix:
-  allow_failures:
-    - rvm: 1.8
-script: ./script/test
```
[the remark this answers is not in the document]
Comment (kiskoza) [reply .travis.yml:13]: I've removed `script/test` :+1:

File: lib/crack/json.rb (hunk 1 above)
Comment (kiskoza) [reply lib/crack/json.rb:7]: Without 644b0f7 / #74 , the CI fails for Ruby 3.1, 3.2, 3.3 and JRuby equivalents, a sample run could be found here: https://github.com/kiskoza/crack/actions/runs/7622992428/job/20762118748

How should we resolve this?
1. Keep the commit fixing the code
2. Keep the Ruby versions within the CI matrix, but merge them as failing
3. Remove the newer Ruby versions and add them along the fix in a new PR

I would prefer 1 or 2 and wouldn't recommend doing 3, but happy to go the path you find the best

Comment (bf4) [reply lib/crack/json.rb:7]: I think simplest is to merge this with passing ci (comment out or allow failures) then in fix pr add the failing matrix options

Comment (kiskoza) [reply lib/crack/json.rb]: I've updated the CI matrix to allow failures for recent Ruby versions, it should be fine now. I'd happily help as being a maintainer as well.

**Comment by bf4** [comment]
@kiskoza You've been granted the respect and responsibility of a maintainer by myself and @jnunemaker :) Thanks for your assistance!

**Pull request merged**